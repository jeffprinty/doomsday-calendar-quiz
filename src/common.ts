import Chance from 'chance';
import { DateTime } from 'luxon';

export const chance = new Chance();

export const correctColor = 'bg-green-600';
export const incorrectColor = 'bg-red-900';

export const stepZero = 'text-pink-400';
export const stepOne = 'text-red-400';
export const stepTwo = 'text-orange-400';
export const stepThree = 'text-yellow-400';
export const stepFour = 'text-green-400';
export const stepFive = 'text-blue-400';
export const stepSix = 'text-indigo-400';
export type Steps = 'stepOne' | 'stepTwo' | 'stepThree' | 'stepFour' | 'stepFive';

export type AnchorDayCentury = '18' | '19' | '20' | '21';

export const anchorDays = {
  '18': 5,
  '19': 3,
  '20': 2,
  '21': 0
};

export const getAnchorDayForCentury = (century: AnchorDayCentury) => {
  return anchorDays[century];
};

export const sharedMemeticHandles = {
  doubles: '4/4, 6/6, 8/8, 10/10, 12/12',
  nineToFive: '9-to-5 at 7-11'
};

export const monthNames = [
  'January',
  'February',
  'March',
  'April',
  'May',
  'June',
  'July',
  'August',
  'September',
  'October',
  'November',
  'December'
];

export const dayAbbreviations = ['Su', 'Mo', 'Tu', 'We', 'Th', 'Fr', 'Sa'];

export type Day = 'Sun' | 'Mon' | 'Tue' | 'Wed' | 'Thu' | 'Fri' | 'Sat';
export const daysOfWeek: Array<Day> = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];

export const dayNames = [
  'Sunday',
  'Monday',
  'Tuesday',
  'Wednesday',
  'Thursday',
  'Friday',
  'Saturday'
];

export const mnemonics = [
  {
    month: 'jan',
    monthName: 'January',
    monthNumber: 1,
    common: 3,
    leap: 4
  },
  {
    month: 'feb',
    monthName: 'February',
    monthNumber: 2,
    common: 28,
    leap: 29
  },
  {
    month: 'mar',
    monthName: 'March',
    monthNumber: 3,
    common: 14,
    leap: 14,
    memeticHandle: 'pi day'
  },
  {
    month: 'apr',
    monthName: 'April',
    monthNumber: 4,
    common: 4,
    leap: 4,
    memeticHandle: sharedMemeticHandles.doubles
  },
  {
    month: 'may',
    monthName: 'May',
    monthNumber: 5,
    common: 9,
    leap: 9,
    memeticHandle: sharedMemeticHandles.nineToFive
  },
  {
    month: 'jun',
    monthName: 'June',
    monthNumber: 6,
    common: 6,
    leap: 6,
    memeticHandle: sharedMemeticHandles.doubles
  },
  {
    month: 'jul',
    monthName: 'July',
    monthNumber: 7,
    common: 11,
    leap: 11,
    memeticHandle: sharedMemeticHandles.nineToFive
  },
  {
    month: 'aug',
    monthName: 'August',
    monthNumber: 8,
    common: 8,
    leap: 8,
    memeticHandle: sharedMemeticHandles.doubles
  },
  {
    month: 'sep',
    monthName: 'September',
    monthNumber: 9,
    common: 5,
    leap: 5,
    memeticHandle: sharedMemeticHandles.nineToFive
  },
  {
    month: 'oct',
    monthName: 'October',
    monthNumber: 10,
    common: 10,
    leap: 10,
    memeticHandle: sharedMemeticHandles.doubles
  },
  {
    month: 'nov',
    monthName: 'November',
    monthNumber: 11,
    common: 7,
    leap: 7,
    memeticHandle: sharedMemeticHandles.nineToFive
  },
  {
    month: 'dec',
    monthName: 'December',
    monthNumber: 12,
    common: 12,
    leap: 12,
    memeticHandle: sharedMemeticHandles.doubles
  }
];

export const guessDateFormat = 'MMMM dd, yyyy';

export const generateDaysTable = () => {
  const firstDayOfMonth = new Date(2025, 1, 1);
  const weekdayOfFirstDay = firstDayOfMonth.getDay();
  const currentDays = [];

  for (let day = 0; day < 360; day++) {
    if (day === 0 && weekdayOfFirstDay === 0) {
      firstDayOfMonth.setDate(firstDayOfMonth.getDate() - 7);
    } else if (day === 0) {
      firstDayOfMonth.setDate(firstDayOfMonth.getDate() + (day - weekdayOfFirstDay));
    } else {
      firstDayOfMonth.setDate(firstDayOfMonth.getDate() + 1);
    }

    const monthNumber = firstDayOfMonth.getMonth();
    const monthNumberForHumans = monthNumber + 1;
    const monthMnemonic = mnemonics[monthNumber];
    const dayIsDoomsday = monthMnemonic.common === firstDayOfMonth.getDay();

    const calendarDay = {
      // currentMonth: firstDayOfMonth.getMonth() === props.day.getMonth(),
      date: new Date(firstDayOfMonth),
      monthNumber,
      monthNumberForHumans,
      number: firstDayOfMonth.getDate(),
      // selected: firstDayOfMonth.toDateString() === props.day.toDateString(),
      year: firstDayOfMonth.getFullYear(),
      dayIsDoomsday
    };

    currentDays.push(calendarDay);
  }
  return currentDays;
};

export const betterDaysTable = (howManyDays = 360) => {
  const startingDay = DateTime.fromISO('2024-12-29');
  const daysArray = [];
  for (let day = 0; day < howManyDays; day++) {
    const dayy = startingDay.plus({ days: day });
    const dayObject = {
      date: dayy,
      month: dayy.get('month'),
      dayNumber: dayy.get('day'),
      cellNumber: day
    };
    daysArray.push(dayObject);
  }
  return daysArray;
};

export const chunkArray = <T>(arrayToChunk: Array<T>, size: number): Array<Array<T>> =>
  Array.from({ length: Math.ceil(arrayToChunk.length / size) }, (v, index) =>
    arrayToChunk.slice(index * size, index * size + size)
  );

export const getRandomDateInYear = (year: number) => {
  const randomDayOfYear = chance.integer({ min: 1, max: 365 });
  return DateTime.fromObject({
    year,
    ordinal: randomDayOfYear
  });
};
