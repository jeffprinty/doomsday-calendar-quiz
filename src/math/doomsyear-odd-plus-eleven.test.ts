import { expect, test } from 'vitest';

import { doomsyearTable } from '../common';
import oddPlusEleven, { oddPlusElevenFull } from './doomsyear-odd-plus-eleven';

for (const [year, result] of doomsyearTable) {
  test(`tests oddPlusEleven for ${year}`, () => {
    expect(oddPlusEleven(year - 2000)).toBe(result);
  });
}
for (const [year] of doomsyearTable) {
  test(`tests oddPlusElevenFull for ${year}`, () => {
    expect(oddPlusElevenFull(year - 2000)).toMatchSnapshot();
  });
}
