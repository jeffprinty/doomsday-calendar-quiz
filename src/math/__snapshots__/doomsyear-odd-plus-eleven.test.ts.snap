// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests oddPlusElevenFull for 2000 1`] = `
{
  "firstResult": 0,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 0,
}
`;

exports[`tests oddPlusElevenFull for 2001 1`] = `
{
  "firstResult": 6,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 6,
}
`;

exports[`tests oddPlusElevenFull for 2002 1`] = `
{
  "firstResult": 1,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 12,
}
`;

exports[`tests oddPlusElevenFull for 2003 1`] = `
{
  "firstResult": 7,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 18,
}
`;

exports[`tests oddPlusElevenFull for 2004 1`] = `
{
  "firstResult": 2,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 2,
}
`;

exports[`tests oddPlusElevenFull for 2005 1`] = `
{
  "firstResult": 8,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 8,
}
`;

exports[`tests oddPlusElevenFull for 2006 1`] = `
{
  "firstResult": 3,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 14,
}
`;

exports[`tests oddPlusElevenFull for 2007 1`] = `
{
  "firstResult": 9,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 20,
}
`;

exports[`tests oddPlusElevenFull for 2008 1`] = `
{
  "firstResult": 4,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 4,
}
`;

exports[`tests oddPlusElevenFull for 2009 1`] = `
{
  "firstResult": 10,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 10,
}
`;

exports[`tests oddPlusElevenFull for 2010 1`] = `
{
  "firstResult": 5,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 16,
}
`;

exports[`tests oddPlusElevenFull for 2011 1`] = `
{
  "firstResult": 11,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 22,
}
`;

exports[`tests oddPlusElevenFull for 2012 1`] = `
{
  "firstResult": 6,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 6,
}
`;

exports[`tests oddPlusElevenFull for 2013 1`] = `
{
  "firstResult": 12,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 12,
}
`;

exports[`tests oddPlusElevenFull for 2014 1`] = `
{
  "firstResult": 7,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 18,
}
`;

exports[`tests oddPlusElevenFull for 2015 1`] = `
{
  "firstResult": 13,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 24,
}
`;

exports[`tests oddPlusElevenFull for 2016 1`] = `
{
  "firstResult": 8,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 8,
}
`;

exports[`tests oddPlusElevenFull for 2017 1`] = `
{
  "firstResult": 14,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 14,
}
`;

exports[`tests oddPlusElevenFull for 2018 1`] = `
{
  "firstResult": 9,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 20,
}
`;

exports[`tests oddPlusElevenFull for 2019 1`] = `
{
  "firstResult": 15,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 26,
}
`;

exports[`tests oddPlusElevenFull for 2020 1`] = `
{
  "firstResult": 10,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 10,
}
`;

exports[`tests oddPlusElevenFull for 2021 1`] = `
{
  "firstResult": 16,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 16,
}
`;

exports[`tests oddPlusElevenFull for 2022 1`] = `
{
  "firstResult": 11,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 22,
}
`;

exports[`tests oddPlusElevenFull for 2023 1`] = `
{
  "firstResult": 17,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 28,
}
`;

exports[`tests oddPlusElevenFull for 2024 1`] = `
{
  "firstResult": 12,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 12,
}
`;

exports[`tests oddPlusElevenFull for 2025 1`] = `
{
  "firstResult": 18,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 18,
}
`;

exports[`tests oddPlusElevenFull for 2026 1`] = `
{
  "firstResult": 13,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 24,
}
`;

exports[`tests oddPlusElevenFull for 2027 1`] = `
{
  "firstResult": 19,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 30,
}
`;

exports[`tests oddPlusElevenFull for 2028 1`] = `
{
  "firstResult": 14,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 14,
}
`;

exports[`tests oddPlusElevenFull for 2029 1`] = `
{
  "firstResult": 20,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 20,
}
`;

exports[`tests oddPlusElevenFull for 2030 1`] = `
{
  "firstResult": 15,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 26,
}
`;

exports[`tests oddPlusElevenFull for 2031 1`] = `
{
  "firstResult": 21,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 32,
}
`;

exports[`tests oddPlusElevenFull for 2032 1`] = `
{
  "firstResult": 16,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 16,
}
`;

exports[`tests oddPlusElevenFull for 2033 1`] = `
{
  "firstResult": 22,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 22,
}
`;

exports[`tests oddPlusElevenFull for 2034 1`] = `
{
  "firstResult": 17,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 28,
}
`;

exports[`tests oddPlusElevenFull for 2035 1`] = `
{
  "firstResult": 23,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 34,
}
`;

exports[`tests oddPlusElevenFull for 2036 1`] = `
{
  "firstResult": 18,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 18,
}
`;

exports[`tests oddPlusElevenFull for 2037 1`] = `
{
  "firstResult": 24,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 24,
}
`;

exports[`tests oddPlusElevenFull for 2038 1`] = `
{
  "firstResult": 19,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 30,
}
`;

exports[`tests oddPlusElevenFull for 2039 1`] = `
{
  "firstResult": 25,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 36,
}
`;

exports[`tests oddPlusElevenFull for 2040 1`] = `
{
  "firstResult": 20,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 20,
}
`;

exports[`tests oddPlusElevenFull for 2041 1`] = `
{
  "firstResult": 26,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 26,
}
`;

exports[`tests oddPlusElevenFull for 2042 1`] = `
{
  "firstResult": 21,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 32,
}
`;

exports[`tests oddPlusElevenFull for 2043 1`] = `
{
  "firstResult": 27,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 38,
}
`;

exports[`tests oddPlusElevenFull for 2044 1`] = `
{
  "firstResult": 22,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 22,
}
`;

exports[`tests oddPlusElevenFull for 2045 1`] = `
{
  "firstResult": 28,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 28,
}
`;

exports[`tests oddPlusElevenFull for 2046 1`] = `
{
  "firstResult": 23,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 34,
}
`;

exports[`tests oddPlusElevenFull for 2047 1`] = `
{
  "firstResult": 29,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 40,
}
`;

exports[`tests oddPlusElevenFull for 2048 1`] = `
{
  "firstResult": 24,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 24,
}
`;

exports[`tests oddPlusElevenFull for 2049 1`] = `
{
  "firstResult": 30,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 30,
}
`;

exports[`tests oddPlusElevenFull for 2050 1`] = `
{
  "firstResult": 25,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 36,
}
`;

exports[`tests oddPlusElevenFull for 2051 1`] = `
{
  "firstResult": 31,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 42,
}
`;

exports[`tests oddPlusElevenFull for 2052 1`] = `
{
  "firstResult": 26,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 26,
}
`;

exports[`tests oddPlusElevenFull for 2053 1`] = `
{
  "firstResult": 32,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 32,
}
`;

exports[`tests oddPlusElevenFull for 2054 1`] = `
{
  "firstResult": 27,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 38,
}
`;

exports[`tests oddPlusElevenFull for 2055 1`] = `
{
  "firstResult": 33,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 44,
}
`;

exports[`tests oddPlusElevenFull for 2056 1`] = `
{
  "firstResult": 28,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 28,
}
`;

exports[`tests oddPlusElevenFull for 2057 1`] = `
{
  "firstResult": 34,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 34,
}
`;

exports[`tests oddPlusElevenFull for 2058 1`] = `
{
  "firstResult": 29,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 40,
}
`;

exports[`tests oddPlusElevenFull for 2059 1`] = `
{
  "firstResult": 35,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 46,
}
`;

exports[`tests oddPlusElevenFull for 2060 1`] = `
{
  "firstResult": 30,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 30,
}
`;

exports[`tests oddPlusElevenFull for 2061 1`] = `
{
  "firstResult": 36,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 36,
}
`;

exports[`tests oddPlusElevenFull for 2062 1`] = `
{
  "firstResult": 31,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 42,
}
`;

exports[`tests oddPlusElevenFull for 2063 1`] = `
{
  "firstResult": 37,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 48,
}
`;

exports[`tests oddPlusElevenFull for 2064 1`] = `
{
  "firstResult": 32,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 32,
}
`;

exports[`tests oddPlusElevenFull for 2065 1`] = `
{
  "firstResult": 38,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 38,
}
`;

exports[`tests oddPlusElevenFull for 2066 1`] = `
{
  "firstResult": 33,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 44,
}
`;

exports[`tests oddPlusElevenFull for 2067 1`] = `
{
  "firstResult": 39,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 50,
}
`;

exports[`tests oddPlusElevenFull for 2068 1`] = `
{
  "firstResult": 34,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 34,
}
`;

exports[`tests oddPlusElevenFull for 2069 1`] = `
{
  "firstResult": 40,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 40,
}
`;

exports[`tests oddPlusElevenFull for 2070 1`] = `
{
  "firstResult": 35,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 46,
}
`;

exports[`tests oddPlusElevenFull for 2071 1`] = `
{
  "firstResult": 41,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 52,
}
`;

exports[`tests oddPlusElevenFull for 2072 1`] = `
{
  "firstResult": 36,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 36,
}
`;

exports[`tests oddPlusElevenFull for 2073 1`] = `
{
  "firstResult": 42,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 42,
}
`;

exports[`tests oddPlusElevenFull for 2074 1`] = `
{
  "firstResult": 37,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 48,
}
`;

exports[`tests oddPlusElevenFull for 2075 1`] = `
{
  "firstResult": 43,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 54,
}
`;

exports[`tests oddPlusElevenFull for 2076 1`] = `
{
  "firstResult": 38,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 38,
}
`;

exports[`tests oddPlusElevenFull for 2077 1`] = `
{
  "firstResult": 44,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 44,
}
`;

exports[`tests oddPlusElevenFull for 2078 1`] = `
{
  "firstResult": 39,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 50,
}
`;

exports[`tests oddPlusElevenFull for 2079 1`] = `
{
  "firstResult": 45,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 56,
}
`;

exports[`tests oddPlusElevenFull for 2080 1`] = `
{
  "firstResult": 40,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 40,
}
`;

exports[`tests oddPlusElevenFull for 2081 1`] = `
{
  "firstResult": 46,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 46,
}
`;

exports[`tests oddPlusElevenFull for 2082 1`] = `
{
  "firstResult": 41,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 52,
}
`;

exports[`tests oddPlusElevenFull for 2083 1`] = `
{
  "firstResult": 47,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 58,
}
`;

exports[`tests oddPlusElevenFull for 2084 1`] = `
{
  "firstResult": 42,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 42,
}
`;

exports[`tests oddPlusElevenFull for 2085 1`] = `
{
  "firstResult": 48,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 48,
}
`;

exports[`tests oddPlusElevenFull for 2086 1`] = `
{
  "firstResult": 43,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 54,
}
`;

exports[`tests oddPlusElevenFull for 2087 1`] = `
{
  "firstResult": 49,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 60,
}
`;

exports[`tests oddPlusElevenFull for 2088 1`] = `
{
  "firstResult": 44,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 44,
}
`;

exports[`tests oddPlusElevenFull for 2089 1`] = `
{
  "firstResult": 50,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 50,
}
`;

exports[`tests oddPlusElevenFull for 2090 1`] = `
{
  "firstResult": 45,
  "moduloFromSeven": 0,
  "moduloResult": 0,
  "secondResult": 56,
}
`;

exports[`tests oddPlusElevenFull for 2091 1`] = `
{
  "firstResult": 51,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 62,
}
`;

exports[`tests oddPlusElevenFull for 2092 1`] = `
{
  "firstResult": 46,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 46,
}
`;

exports[`tests oddPlusElevenFull for 2093 1`] = `
{
  "firstResult": 52,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 52,
}
`;

exports[`tests oddPlusElevenFull for 2094 1`] = `
{
  "firstResult": 47,
  "moduloFromSeven": 5,
  "moduloResult": 2,
  "secondResult": 58,
}
`;

exports[`tests oddPlusElevenFull for 2095 1`] = `
{
  "firstResult": 53,
  "moduloFromSeven": 6,
  "moduloResult": 1,
  "secondResult": 64,
}
`;

exports[`tests oddPlusElevenFull for 2096 1`] = `
{
  "firstResult": 48,
  "moduloFromSeven": 1,
  "moduloResult": 6,
  "secondResult": 48,
}
`;

exports[`tests oddPlusElevenFull for 2097 1`] = `
{
  "firstResult": 54,
  "moduloFromSeven": 2,
  "moduloResult": 5,
  "secondResult": 54,
}
`;

exports[`tests oddPlusElevenFull for 2098 1`] = `
{
  "firstResult": 49,
  "moduloFromSeven": 3,
  "moduloResult": 4,
  "secondResult": 60,
}
`;

exports[`tests oddPlusElevenFull for 2099 1`] = `
{
  "firstResult": 55,
  "moduloFromSeven": 4,
  "moduloResult": 3,
  "secondResult": 66,
}
`;
